### PR TITLE
feat(classification): backport Ruby and C# test cases and KPIs

### PR DESCRIPTION
**Pull request by elsapet**
## Description

For Ruby and C# cases: 

- Deduplicate input and output CSVs into a JSON file, and include object classifications where missing. 
- Add test helper to construct expected and actual Classification objects and KPIs from JSON file
- Snapshot the results - actual valid classifications and KPIs
- Some fixes to the algorithm e.g. 
   - Exclude objects with blank names
   - For unknown objects, check fields for DB identifiers (we updated the Miro flow to include this edge case)

Still TODO (later PR)
- [ ] Other languages

## Checklist

- [ ] I've added test coverage that shows my fix or feature works as expected.
- [ ] I've updated or added documentation if required.
- [ ] I've included usage information in the description if CLI behavior was updated or added.
- [ ] PR title follows [Conventional Commits](https://www.conventionalcommits.org/) format

### CODE REVIEW

File: pkg/classification/schema/fixtures/ruby.json [hunk 1]
```diff
@@ -0,0 +1,1362 @@
...
+      {
+        "name": "customerid",
+        "type": "unknown",
+        "data_type_info": "Customer",
+        "state": "valid",
+        "reason": "known_database_identifier",
+        "false_positive": false
+      }
+    ],
+    "state": "invalid"
```
Comment (elsapet) [inline pkg/classification/schema/fixtures/ruby.json:72]: Manual intervention here: this is an invalid classification - where is not a recognised object pattern, despite its valid field "customerid"

/cc @cfabianski does it sound right to you?

**Review by cfabianski — DISMISSED** [review]

**Comment by elsapet** [comment]
@cfabianski could you re-approve when you've a moment? I rebased, sorry! 🙈

**Review by cfabianski — APPROVED** [review]
